### PR TITLE
Add fasm2bels for PP3

### PR DESCRIPTION
**Pull request by mkurc-ant**
This PR enables fasm2bels for PP3 both in the CMake flow and in installed toolchain.

There are other minor fixes / improvements included:
 - Fixed SDC dependency issue in CMake
 - Added a check to `create_ioplace` which prevents from constraining multiple pins to the same pad.

### CODE REVIEW

File: .github/workflows/quicklogic.yml [hunk 1]
```diff
@@ -70,7 +70,7 @@ jobs:
           export PATH=$PATH:$PWD/.tmp
           export ROOT=$(pwd)
           export INSTALL_DIR=quicklogic-arch-defs
-          export CMAKE_FLAGS="-GNinja -DINSTALL_FAMILIES=qlf_k4n8,qlf_k6n10,pp3 -DCMAKE_INSTALL_PREFIX=${ROOT}/${INSTALL_DIR}"
+          export CMAKE_FLAGS="-GNinja -DINSTALL_FAMILIES=pp3 -DCMAKE_INSTALL_PREFIX=${ROOT}/${INSTALL_DIR}"
```
Comment (acomodi) [inline .github/workflows/quicklogic.yml:73]: Why qlf devices are being removed from the install package?

File: quicklogic/pp3/boards.cmake [hunk 2]
```diff
@@ -4,6 +4,7 @@ add_quicklogic_board(
   DEVICE ql-eos-s3
   PACKAGE wlcsp
   FABRIC_PACKAGE PD64
+  BIT_TO_V_EXTRA_ARGS "--device-name eos-s3 --package-name PD64"
```
Comment (acomodi) [inline quicklogic/pp3/boards.cmake:7]: Cannot these args be inferred by other args? e.g. `FABRIC_PACKAGE` and `DEVICE`

**Review by acomodi — APPROVED** [review]
LGTM, left some comments

File: .github/workflows/quicklogic.yml (hunk 1 above)
Comment (mkurc-ant) [reply .github/workflows/quicklogic.yml:73]: This is only temporary. I'm investigating a CI failure I couldn't reproduce locally.

File: quicklogic/pp3/boards.cmake (hunk 2 above)
Comment (mkurc-ant) [reply quicklogic/pp3/boards.cmake]: They can. I have fixed that.

**Review by lnsharma — APPROVED** [review]
LGTM

**Pull request merged**